src/text-editor.js

```diff
@@ -4169,20 +4156,33 @@ module.exports = class TextEditor {
     return this.preferredLineLength;
   }
 
-  // Essential: Toggle soft wrapping for this editor
+  // Essential: Disables soft wrapping for this editor
   //
   // Returns a {Boolean}.
-  toggleSoftWrapped() {
-    return this.setSoftWrapped(!this.isSoftWrapped());
+  disableSoftWrapped() {
+    return this.setSoftWrapped('disabled');
   }
 
   // Essential: Gets the column at which column will soft wrap
   getSoftWrapColumn() {
     if (this.isSoftWrapped() && !this.mini) {
-      if (this.softWrapAtPreferredLineLength) {
-        return Math.min(this.getEditorWidthInChars(), this.preferredLineLength);
-      } else {
-        return this.getEditorWidthInChars();
+      switch (this.softWrapped) {
+        case 'window':
+          return this.getEditorWidthInChars();
+        case 'preferredLineLength':
+          return this.preferredLineLength;
+        case 'minWindowOrPll':
```

Comment:
```suggestion
        case 'minWindowOrPreferredLineLength':
```
I know this is a long one but it is clear on what the option is supposed to achieve.